apps/emqx/src/emqx_router.erl

```diff
@@ -440,38 +437,20 @@ has_route_v2(Topic, Dest) ->
     end.
 
 cleanup_routes_v2(Node) ->
-    % NOTE
-    % No point in transaction here because all the operations on filters table are dirty.
-    ok = ets:foldl(
-        fun(#routeidx{entry = K}, ok) ->
-            case get_dest_node(emqx_topic_index:get_id(K)) of
-                Node ->
-                    mria:dirty_delete(?ROUTE_TAB_FILTERS, K);
-                _ ->
-                    ok
-            end
-        end,
-        ok,
-        ?ROUTE_TAB_FILTERS
-    ),
-    ok = ets:foldl(
-        fun(#route{dest = Dest} = Route, ok) ->
-            case get_dest_node(Dest) of
-                Node ->
-                    mria:dirty_delete_object(?ROUTE_TAB, Route);
-                _ ->
-                    ok
-            end
+    lists:foreach(
```

Comment:
Note: both v1 and v2 cleanups are backward compatible at Mnesia level: older nodes that don't have Mnesia patch will commit the transaction correctly (thanks to the fact that Mnesia internally was ready for match_delete implementation). 
Older Mria releases would fail on both core and replicant nodes (even though Mria ptotcol version is bumped and new replicant nodes will only connect to new cores), However, the failure will lead to shard restart and re-bootstrapping which  will eventually bring all nodes to the consistent state.